### PR TITLE
User Management Redesign and User Details Page

### LINKED ISSUE #8878: Refactor user list and user management pages
> [!NOTE] 
> Create wireframes for the new user list and details page before asking for the issue to be assigned

Current user list page:
Image

Required changes:

- [ ] Extract user cards as a separate component
- [ ] Clicking the user card should bring up the user detail page
- [ ] Move the action buttons like delete, manage linked facilities/skills, set working hours to the user detail page
- [ ] Add user avatars to user cards, similar to user card from personal details page of the logged-in user
Image

- Fix the alignment of the search bar and action buttons in desktop mode

New User detail page:

TODO: add wireframes

Required changes:

- [ ] Convert linked facility/skills management to a dropdown within the details page
Image

### LINKED ISSUE #8983: Enhance Grid Layout in "View Users Associated with Facilities" Section
The current grid layout in the "View Users" section under facility management appears cluttered and could benefit from a more organized structure. A switch to a list-like layout may improve readability and user experience, providing a cleaner and more intuitive interface for viewing user details.

Before:
Image
Steps to Reproduce:

Go to Facilities.
Click on View Facility for a specific facility.
Navigate to Manage Facility.
Select View Users.
Observed Issue: The grid layout in the "View Users" section is somewhat cluttered, making it harder for users to scan through information quickly.

Proposed Enhancement: Consider changing the grid layout to a list-style view. 

Improve readability by reducing visual clutter.
Potentially allow for more user-friendly sorting and filtering options in the future.

### LINKED ISSUE #9001: Improve Age Validation in User Creation Form
**Description**:Currently, when creating a user, if the Date of Birth (DOB) entered corresponds to an age less than 15 years, the system makes an API call and returns an error with the message: "Age must be greater than 15 years."

Image

**Proposed Improvement:** To enhance the user experience and reduce unnecessary API calls, dynamic client-side validation should be added to the form. This validation would check if the age is at least 15 years before submitting the form, thereby providing immediate feedback to the user.

**Expected Behavior:**

The form should display an error message dynamically if the user enters a DOB indicating an age less than 15 years.
The form should prevent the API call from being made if the age criteria are not met.
**Benefits**:

Reduces API traffic.
Improves user experience with real-time feedback.

### PR DESCRIPTION
**Pull request by Jacobjeevan**
## Proposed Changes

- Fixes #8878
- Fixes #9001 
- Fixes #8983
- BE PR https://github.com/ohcnetwork/care/pull/2595

### To Do:

- Open an issue for moving Profile: I've already moved most of the elements from existing profile page to to user details page (when it's details of logged in user), but there's a few issues in the works modifying certain elements in the profile page, so may need to tweak user details page.
- Done: While testing, I noticed that user deletion doesn't work for facility users page (delete route sets `is_active flag` to false, which works fine for userList (`/api/v1/users/`), but not for getFacilityUsers (`facility/facilityId/get_users/`) as latter is checking for `deleted=False` flag. BE PR [here](https://github.com/ohcnetwork/care/pull/2638)

### Potential things to check:

- Conditional rendering of elements
   - Qualifications (under Professional Information) only for Doctors/Nurses
   - YOE and Medical registration number (under Professional Information) only for Doctors
   - Logged in user can edit their own profile as well as change password
   - Edit (other users) only available for District/StateAdmin and superusers only
   - Deletion only (other users) for District/StateAdmins and superusers only
       - Only can delete users below in hierarchy
   - Avatar edit option (for all users) only for super user
   - Logged in user can edit their own avatar
- Users without edit permissions can view profiles of other users

image

image

image

image

image

image

image

image

image

image

If user doesn't have edit permissions:

image

image

@ohcnetwork/care-fe-code-reviewers

## Merge Checklist

- [ ] Add specs that demonstrate bug / test a new feature.
- [ ] Update [product documentation](https://docs.ohc.network).
- [ ] Ensure that UI text is kept in I18n files.
- [x] Prep screenshot or demo video for changelog entry, and attach it to issue.
- [ ] Request for Peer Reviews
- [ ] Completion of QA

## Release Notes

- **New Features**
	- Enhanced user management functionalities with new localization entries for account actions.
	- Introduced components for managing linked facilities and skills, including confirmation dialogs.
	- Added user avatar management capabilities, allowing users to upload and delete profile pictures.
	- Implemented a comprehensive user editing interface with toggling between view and edit modes.
	- Added a new component for displaying linked skills associated with a user.
	- Introduced a new component for managing user summaries and interactions.
	- Added a new `LinkedSkillsTab` component for displaying linked skills associated with a user.
	- Introduced a `UserHome` component for user management with dynamic tab rendering.
	- Added a `UserBanner` component for displaying user information in a banner format.
	- Introduced a `UserResetPassword` component for managing password resets.
	- Enhanced error handling and validation logic across various user management components.

- **Bug Fixes**
	- Improved validation logic for user forms, ensuring accurate error messages and handling.
	- Streamlined state management in user-related components for better performance.

- **Documentation**
	- Updated internal documentation to reflect new props and components introduced in this release.

- **Tests**
	- Added new end-to-end tests for user homepage functionalities and user management scenarios.
	- Updated existing tests to reflect changes in error messages and selectors for user verification.

### CODE REVIEW

File: src/components/Users/LinkedFacilitiesTab.tsx [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import userColumns from "@/components/Common/UserColumns";
+import LinkedFacilities from "@/components/Users/LinkedFacilities";
+import { UserModel } from "@/components/Users/models";
+
+type Props = {
+  userData: UserModel;
+  username: string;
+  refetchUserData?: () => void;
+};
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/LinkedFacilitiesTab.tsx:17]: Can't remove that as it's being passed as a prop to UserColumns function.

File: src/components/Users/LinkedFacilities.tsx [hunk 2]
```diff
@@ -0,0 +1,303 @@
...
+};
+
+export default function LinkedFacilities({
+  userData,
+  refetchUserData,
+}: {
+  userData: UserModel;
+  refetchUserData?: () => void;
+}) {
+  const [facility, setFacility] = useState<any>(null);
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/LinkedFacilities.tsx:46]: Best to keep it that way as facilitySelect supports array of FacilityModel items, however we are restricting selection to just one in LinkedFacilities.

File: src/components/Users/UserListAndCard.tsx [hunk 3]
```diff
@@ -0,0 +1,374 @@
...
+    <>
+      <div className="mb-4 flex flex-col items-center justify-between gap-3 sm:flex-row">
+        <div className="sm:w-1/2">
+          <SearchInput
+            id="search-by-username"
+            name="username"
+            onChange={(e) => onSearch(e.value)}
+            value={searchValue}
+            placeholder={t("search_by_username")}
+          />
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserListAndCard.tsx:372]: Already handled elsewhere (displays a no results banner if there aren't any results and searchbox is reset on clearing of the badge).

File: src/components/Users/UserAddEditForm.tsx [hunk 4]
```diff
@@ -0,0 +1,1232 @@
...
+        msg: "User added successfully",
+      });
+      navigate("/users");
+    } else {
+      Notification.Error({
+        msg: error?.message ?? "Error while adding user",
+      });
+    }
+    setIsLoading(false);
+  };
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserAddEditForm.tsx]: You have already mentioned this in a previous comment.

File: src/Utils/permissions.ts [hunk 5]
```diff
@@ -10,21 +31,30 @@ export const showUserDelete = (authUser: UserModel, targetUser: UserModel) => {
...
-    authUser.user_type === "DistrictAdmin" &&
-    targetUser.district_object?.id === authUser.district
+    USER_TYPES.indexOf(authUser.user_type) <=
+    USER_TYPES.indexOf(targetUser.user_type)
   )
-    return true;
+    return false;
 
-  return false;
+  return checkIfStateOrDistrictAdminInSameLocation(authUser, targetUser);
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/Utils/permissions.ts]: No, both of those objects should be present when showUserPasswordReset is called.

File: cypress/pageobject/Users/ManageUserPage.ts [hunk 6]
```diff
@@ -68,6 +64,38 @@ export class ManageUserPage {
...
+    email: string,
+    dateOfBirth: string,
+    gender: string,
+  ) {
+    cy.get("#first_name").should("have.value", fName);
+    cy.get("#last_name").should("have.value", lName);
+    cy.get("#email").should("have.value", email);
+    cy.get("#date_of_birth").should("have.value", dateOfBirth);
+    cy.get("#gender").should("contain.text", gender);
+  }
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply cypress/pageobject/Users/ManageUserPage.ts]: Unnecessary, as the method is called to check input provided earlier in the test.

File: src/components/Users/UserListAndCard.tsx [hunk 7]
```diff
@@ -0,0 +1,413 @@
...
+  user: UserModel | UserAssignedModel,
+  cur_online: boolean,
+  showDetailsButton = false,
+) => {
+  return (
+    <div>
+      <div className="flex flex-row justify-between gap-x-3">
+        <div className="flex flex-col">
+          <div className="flex items-center gap-x-3">
+            <h1 id={`name-${user.username}`} className="text-base font-bold">
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserListAndCard.tsx:84]: username is unique.

File: src/components/Users/UserListAndCard.tsx [hunk 8]
```diff
@@ -0,0 +1,413 @@
...
+      // Temporarily allows creation of users with elevated permissions due to introduction of new roles.
+      return [...defaultAllowedUserTypes, USER_TYPE_OPTIONS[6]];
+    default:
+      return defaultAllowedUserTypes;
+  }
+};
+export const CanUserAccess = (user: UserModel | UserAssignedModel) => {
+  const allowedTypes = GetUserTypes(true).map((type) => type.id);
+  return allowedTypes.includes(user.user_type);
+};
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserListAndCard.tsx:58]: User type is expected to be present for all users.

File: cypress/pageobject/Users/ManageUserPage.ts [hunk 9]
```diff
@@ -116,10 +152,45 @@
...
+
+  verifyDoctorQualification() {
+    cy.get("input[name='qualification']").should("be.visible");
+  }
+
+  verifyLinkedSkillsTabPage() {
+    cy.get("#select-skill").scrollIntoView();
+    cy.get("#select-skill").should("be.visible");
+  }
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply cypress/pageobject/Users/ManageUserPage.ts]: Not needed as the method is called specifically to verify the content for Doctor roles, however a similar method could be added to verify the element not occurring for other roles.

File: cypress/e2e/users_spec/UsersManage.cy.ts [hunk 10]
```diff
@@ -85,13 +121,17 @@ describe("Manage User", () => {
...
+    manageUserPage.clickMoreDetailsButton(usernameforworkinghour);
+    manageUserPage.verifyMoreDetailsPage();
+    manageUserPage.verifyProfileTabPage();
     manageUserPage.clearweeklyhourfield();
     manageUserPage.clickSubmit();
-    manageUserPage.verifyErrorText("Value should be between 0 and 168");
-    // verify the data is reflected in user card and profile page
+    manageUserPage.verifyErrorText(
+      "Average weekly working hours must be a number between 0 and 168",
+    );
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply cypress/e2e/users_spec/UsersManage.cy.ts:301]: Not needed as testing is done with English.

**Comment by Jacobjeevan** [comment]
Cypress test 4 would only work once BE changes are merged (addition of new route).

**Comment by nihal467** [comment]
@Jacobjeevan Saw ongoing discussions on the User Management page design in Slack with code pushes and the Cypress test failures. Once the PR is frozen, please ensure all issues with cypress are resolved and mark it as "Needs Testing."

File: src/components/Users/UserFormValidations.tsx [hunk 11]
```diff
@@ -0,0 +1,148 @@
...
+  password?: string;
+  c_password?: string;
+  facilities?: Array<string>;
+  home_facility?: FacilityModel | null;
+  username?: string;
+  first_name: string;
+  last_name: string;
+  email: string;
+  phone_number: string;
+  alt_phone_number: string;
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserFormValidations.tsx:19]: Additional phone number validation is being worked on in another issue.

File: src/components/Users/UserFormValidations.tsx [hunk 12]
```diff
@@ -0,0 +1,148 @@
...
+  translator: TFunction,
+) => {
+  if (
+    formData.user_type === "Doctor" &&
+    !formData["doctor_medical_council_registration"]
+  ) {
+    return translator("medical_council_registration_required");
+  }
+  return null;
+};
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserFormValidations.tsx]: Is there a known format for this? Presumably we would want to accept registration numbers from around the world.

File: src/components/Users/UserEditDetails.tsx [hunk 13]
```diff
@@ -0,0 +1,177 @@
...
+      />
+      {isEditing ? (
+        <UserAddEditForm
+          username={username}
+          includedFields={editBasicInfoFields}
+          onSubmitSuccess={() => {
+            setIsEditing(false);
+            onSubmitSuccess?.();
+          }}
+        />
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserEditDetails.tsx:97]: Already being handled in UserAddEditForm.

File: src/components/Users/UserListAndCard.tsx [hunk 14]
```diff
@@ -0,0 +1,407 @@
...
+        {user.home_facility_object?.name || t("no_home_facility")}
+      </td>
+      {showDistrictColumn && (
+        <td id="district" className="px-4 py-4 text-sm">
+          {"district_object" in user && user.district_object
+            ? user.district_object?.name
+            : "district" in user && user.district
+              ? user.district
+              : ""}
+        </td>
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserListAndCard.tsx:314]: No, as getDistrict is returning a JSX Element that's styled according to the card view.

File: cypress/pageobject/Users/UserSearch.ts [hunk 15]
```diff
@@ -23,7 +23,7 @@ export class UserPage {
   }
 
   checkUsernameText(username: string) {
-    cy.get(this.usernameText).should("contain.text", username);
+    cy.get(`${this.usernameText}-${username}`).should("contain.text", username);
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply cypress/pageobject/Users/UserSearch.ts:26]: No, this is checking for username in a different page.

File: src/components/Users/UserFormValidations.tsx [hunk 16]
```diff
@@ -0,0 +1,161 @@
...
+      return translator("invalid_url_http_https");
+    }
+    if (parsed.href.toLowerCase().includes("javascript:")) {
+      return translator("invalid_url_javascript");
+    }
+  } catch {
+    return translator("invalid_url");
+  }
+  return null;
+};
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserFormValidations.tsx:161]: No.

File: src/components/Users/UserFormValidations.tsx [hunk 17]
```diff
@@ -0,0 +1,161 @@
...
+    } else if (
+      !validateNumber(formData["doctor_experience_commenced_on"] ?? "") ||
+      Number(formData["doctor_experience_commenced_on"]) < 0 ||
+      Number(formData["doctor_experience_commenced_on"]) > 100
+    ) {
+      return translator("doctor_experience_error");
+    }
+  }
+  return null;
+};
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserFormValidations.tsx:128]: Not needed, what's there now is enough.

File: src/components/Users/UserSummary.tsx [hunk 18]
```diff
@@ -0,0 +1,221 @@
...
+      });
+      setshowDeleteDialog(!showDeleteDialog);
+      navigate("/users");
+    } else {
+      Notification.Error({
+        msg: t("user_delete_error") + ": " + (error || ""),
+      });
+      setshowDeleteDialog(!showDeleteDialog);
+    }
+  };
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply src/components/Users/UserSummary.tsx:71]: If there's an error (most likely some sort of backend/network issue), notification is shown anyways. Closing the delete dialog so user doesn't immediately try to delete the user again.

File: cypress/pageobject/Users/ManageUserPage.ts [hunk 19]
```diff
@@ -1,10 +1,6 @@
 export class ManageUserPage {
   assertHomeFacility(expectedText: string) {
-    cy.get("#home_facility").should("contain.text", expectedText);
-  }
-
-  clickFacilitiesTab() {
-    cy.get("#facilities").click();
+    cy.get("#home-facility").should("contain.text", expectedText);
```
[the remark this answers is not in the document]
Comment (Jacobjeevan) [reply cypress/pageobject/Users/ManageUserPage.ts:3]: don't worry about it.

**Comment by Jacobjeevan** [comment]
Minor changes to the notes displayed (ex: "View or update user's personal information" -> "View user's personal information" if the user has no edit permissions).

Ready for review and testing.

**Comment by nihal467** [comment]
@Jacobjeevan can you clear the merge conflict, its a huge conflict

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan can you clear the merge conflict, its a huge conflict

Done @nihal467

**Comment by nihal467** [comment]
**Test Report**

**Functionality Bug**

image

- [ ] When you search for a username, the input field disappears. It only reappears after clearing the badge. Please revert this to the original behavior where the search field remains visible at all times. 
- [ ] Similar issue noticeable for advanced filter as well 

Note: This behavior is noticeable when we search for no result input 

image

- [ ] Disable the "Add Facility" button for non-admin users on their own pages.
- [ ] Allow non-admin users to unlink their home facility and any facility in their access.

The above-mentioned points are the existing behavior of our platform. Please revert the unintentional changes in your PR and restore the original functionality.

**Minor Bug**

image

- [ ]  The avatar on the user profile page and the profile icon are different. Please make them consistent.

image

- [ ] Reset the component's state after the password has been successfully changed.

**Question For Clarity**

image

- [ ] Only the account owner should be able to edit their own user information. However, with your current PR, an admin can edit other users' profiles. Is this the intended behavior?

image

- [ ] The new profile view details page and the personal profile page display nearly identical data. Could we consider combining them into a single page for simplicity and consistency? [Discuss with the core team].

**Enhancement Issues**

image

- [ ] Ensure the first column remains fixed in the mobile view for the user list. This will improve accessibility and ease of navigation.

image

- [ ] Add a "Last Modified" column in the user view page to display the date and time when the profile was last updated. Additionally, consider implementing an alert system to notify users to reverify their profile details after a period of 6 months or 1 year. This will help ensure that user information remains up-to-date. [Discussion required with core team before proceeding]

**Comment by nihal467** [comment]
@Jacobjeevan check the test report, clear the conflict and attach a screenshot cypress passing in local

**Comment by Jacobjeevan** [comment]
> **Test Report**
> 
> **Functionality Bug**
* [x]  When you search for a username, the input field disappears. It only reappears after clearing the badge. Please revert this to the original behavior where the search field remains visible at all times.
* [x]  Similar issue noticeable for advanced filter as well
* [x]  Disable the "Add Facility" button for non-admin users on their own pages.
* [x]  Allow non-admin users to unlink their home facility and any facility in their access.
* [x]   The avatar on the user profile page and the profile icon are different. Please make them consistent.
* [x]  Reset the component's state after the password has been successfully changed.

> **Question For Clarity**
>     * [ ]  Only the account owner should be able to edit their own user information. However, with your current PR, an admin can edit other users' profiles. Is this the intended behavior?

Yep, that's the intention. 

>     * [ ]  The new profile view details page and the personal profile page display nearly identical data. Could we consider combining them into a single page for simplicity and consistency? [Discuss with the core team].

Yep, we are merging those (removing current profile page), I was waiting on it since there was an intern PR open updating elements in Profile page. We can open an issue that interns can take up once this is merged

* [x]  Ensure the first column remains fixed in the mobile view for the user list. This will improve accessibility and ease of navigation.

>     * [ ]  Add a "Last Modified" column in the user view page to display the date and time when the profile was last updated. Additionally, consider implementing an alert system to notify users to reverify their profile details after a period of 6 months or 1 year. This will help ensure that user information remains up-to-date. [Discussion required with core team before proceeding]

🤔  Yea could do for enhancement.

Here are the tests passing locally:

image

Details on the error (screenshot below). I sometimes encounter this when testing after resetting the db; bypass it by manually logging in the cypress window.

image

File: src/Utils/request/api.tsx [hunk 20]
```diff
@@ -1001,7 +1001,7 @@ const routes = {
   },
 
   getUserDetails: {
-    path: "/api/v1/users/{username}/",
+    path: "/api/v1/users/{username}",
```
Comment (rithviknishad) [inline src/Utils/request/api.tsx]: wouldn't this cause a http 301/302 redirect?

File: src/common/validation.tsx [hunk 21]
```diff
@@ -57,6 +57,11 @@ export const validatePincode = (pincode: string) => {
   return pattern.test(pincode);
 };
 
+export const validateNumber = (number: string) => {
+  const pattern = /^[0-9]+$/;
+  return pattern.test(number);
+};
+
```
Comment (rithviknishad) [inline src/common/validation.tsx:64]: would this allow negative numbers?

also can we could use zod validators for newer forms.

File: src/components/Common/UserColumns.tsx [hunk 22]
```diff
@@ -0,0 +1,31 @@
+import { UserModel } from "../Users/models";
+
+export type userChildProps = {
+  userData: UserModel;
+  username: string;
+  refetchUserData?: () => void;
+};
+
+export default function userColumns(
```
Comment (rithviknishad) [inline src/components/Common/UserColumns.tsx]: We can make this a react component and args could be passed as props right?

```suggestion
export default function UserColumns(
```

**Comment by nihal467** [comment]
@Jacobjeevan the user management test are failing, fix it before requesting a review

File: src/components/Common/UserDetails.tsx [hunk 23]
```diff
@@ -7,8 +7,8 @@ function UserDetails(props: {
 }) {
   return (
     <div className="mt-2" id={props.id}>
-      <div className="font-light leading-relaxed text-secondary-900">
-        {props.title}:
+      <div className="text-sm font-light leading-relaxed text-[#B9B9B9]">
```
Comment (rithviknishad) [inline src/components/Common/UserDetails.tsx]: arbitrary colours are discouraged. let's stick to the design system. use one of gray shade color.

File: src/components/Facility/FacilityUsers.tsx [hunk 24]
```diff
@@ -52,256 +27,44 @@ export default function FacilityUsers(props: any) {
...
-        if (res?.status === 204) {
-          Notification.Success({
-            msg: t("user_deleted_successfuly"),
-          });
-        }
+  const { data: userListData, loading: userListLoading } = useQuery(
+    routes.getFacilityUsers,
+    {
+      query: {
+        limit: resultsPerPage.toString(),
```
Comment (rithviknishad) [inline src/components/Facility/FacilityUsers.tsx]: `request` used by useQuery will convert to string for you.

```suggestion
        limit: resultsPerPage,
```

**Comment by nihal467** [comment]
image

- [ ] the position of badges doesn't look good, we need to think of a different position , which can be taken as a separate issue

**Comment by nihal467** [comment]
conditonally approving it, @khavinshankar @rithviknishad merge only after relevant backend is merged and the cypress is passing

**Review by rithviknishad — COMMENTED** [review]
We could've used shadcn's components and shadcn's form with zod validators in re-designs. But let's do that as separate PR since this is tested.

**Comment by Jacobjeevan** [comment]
> image
> 
>     * [ ]  the position of badges doesn't look good, we need to think of a different position , which can be taken as a separate issue

How's this?

image

File: src/common/validation.tsx (hunk 21 above)
Comment (Jacobjeevan) [reply src/common/validation.tsx:64]: > would this allow negative numbers?
> 
> also can we could use zod validators for newer forms.

Nah, only positive numbers

File: src/Utils/request/api.tsx (hunk 20 above)
Comment (Jacobjeevan) [reply src/Utils/request/api.tsx]: > wouldn't this cause a http 301/302 redirect?

🤔  Doesn't seem to be causing any issues, I'll add it though.

File: src/components/Common/UserColumns.tsx (hunk 22 above)
Comment (Jacobjeevan) [reply src/components/Common/UserColumns.tsx]: Yep, can do.

**Comment by nihal467** [comment]
@Jacobjeevan what is the status on this PR

**Comment by Jacobjeevan** [comment]
> @Jacobjeevan what is the status on this PR

@nihal467 Still waiting on the backend 👍  I'll get the conflicts addressed though.

**Review by rithviknishad — APPROVED** [review]

**Review by khavinshankar — APPROVED** [review]

**Pull request merged**